File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> s1;
        int n = s.size();
        int res = 0;
        int left = 0;
        for (int right = 0; right < n; right++){
            while(s1.count(s[right])){
                s1.erase(s[left]);
                left++;
            }
            s1.insert(s[right]);
            res = max(res, right-left+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let numSet = new Set([...nums]);
        let res = 0;
        for (let num of numSet){
           if(!numSet.has(num - 1)){ // điều kiện này để kiểm tra xem nếu số num hiện tại không phải số đầu của dãy thì ko đếm đỡ mất thời gian
                let count = 1;
                while (numSet.has(num + count)){
                    count++;
                }
                res = Math.max(res, count)
           }
        }
        return res;
    }
}
